src/transport_picoquic.cpp

```diff
@@ -491,8 +493,9 @@ PicoQuicTransport::start()
         logger->info << "Connecting to server " << serverInfo.host_or_ip << ':' << serverInfo.port
                      << std::flush;
 
-        cid = createClient();
-        picoQuicThread = std::thread(&PicoQuicTransport::client, this, cid);
+        if ((cid = createClient())) {
```

Comment:
I was curious about the return type here. Given the return type is a pointer that is typecast, a safer definition for `TransportContextId` would be `uintptr_t`. That is guaranteed to be an integer type of proper size to hold a pointer regardless of the processor's word size.

Reply:
We are returning a copy of the `uni64_t`, which will always be zero or some positive `uin64_t`. **TransportConnId** is a `uint64_t` that is the 64bit pointer memory address value. It is **not used** to def back to a pointer.  It's only used as the numeric value for the connection Id. `reinterpret_cast<TransportConnId>(pq_cnx) is only converting it to a persistent number that can be used as an id.